### ISSUE TITLE
MockHandler with multi-region clients

### ISSUE THREAD

**Issue by pet0ruk** [aws/aws-sdk-php#2233]
Confirm by changing [ ] to [x] below:
- [x] I've gone though [Developer Guide](https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/welcome.html) and [API reference](https://docs.aws.amazon.com/aws-sdk-php/v3/api/index.html)
- [x] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-php-sdk) for answers

**Version of AWS SDK for PHP?**

v3.178.3

**Version of PHP (`php -v`)?**

7.4.9

**Describe the question**

I'm struggling to find any docs on this and can't seem to get it to work. Is it possible to use MockHandler at all with multi region clients?

This test code:

```php
        $sdk = new \Aws\Sdk();
        $mockHandlerNormal = new MockHandler();
        $mockHandlerNormal->append(new Result(["foo" => "bar"]));
        $mockHandlerMultiRegion = new MockHandler();
        $mockHandlerMultiRegion->append(new Result(["foo" => "bar"]));

        $normalS3Client = $sdk->createS3(["region" => "us-east-1", "version" => "latest"]);
        $multiRegionS3Client = $sdk->createMultiRegionS3(["region" => "us-east-1"]);

        $normalS3Client->getHandlerList()->setHandler($mockHandlerNormal);
        $multiRegionS3Client->getHandlerList()->setHandler($mockHandlerMultiRegion);

        var_dump($normalS3Client->listBuckets());
        var_dump($multiRegionS3Client->listBuckets());
```

Results in the following output:

```
object(Aws\Result)#78 (2) {
  ["data":"Aws\Result":private]=>
  array(2) {
    ["foo"]=>
    string(3) "bar"
    ["@metadata"]=>
    array(3) {
      ["effectiveUri"]=>
      string(25) "https://s3.amazonaws.com/"
      ["statusCode"]=>
      int(200)
      ["transferStats"]=>
      array(1) {
        ["http"]=>
        array(1) {
          [0]=>
          array(0) {
          }
        }
      }
    }
  }
  ["monitoringEvents":"Aws\Result":private]=>
  array(0) {
  }
}

PHP Fatal error:  Uncaught ArgumentCountError: Too few arguments to function Aws\MockHandler::__invoke(), 1 passed in dir/vendor/aws/aws-sdk-php/src/S3/S3MultiRegionClient.php on line 264 and exactly 2 expected in dir/vendor/aws/aws-sdk-php/src/MockHandler.php:76
Stack trace:
#0 dir/vendor/aws/aws-sdk-php/src/S3/S3MultiRegionClient.php(264): Aws\MockHandler->__invoke()
#1 [internal function]: Aws\S3\S3MultiRegionClient->Aws\S3\{closure}()
#2 dir/vendor/guzzlehttp/promises/src/Coroutine.php(71): Generator->current()
#3 dir/vendor/guzzlehttp/promises/src/Coroutine.php(86): GuzzleHttp\Promise\Coroutine->__construct()
#4 dir/vendor/guzzlehttp/promises/src/functions.php(362): GuzzleHttp\Promise\Coroutine::of()
#5 dir in dir/vendor/aws/aws-sdk-php/src/MockHandler.php on line 76
```

**Comment by stobrien89** [comment]
Hi @pet0ruk,

Sorry for the delayed response! And thanks for pointing this out. I was able to reproduce. marking this as a bug.  Short answer is no, it's not possible to use a MockHandler with multi-region clients at this time. 

The multi-region client is a special case in that it creates a base client under the hood and makes its requests through that— what's happening here is that we're not able to pass a request (one of two required arguments) to the MockHandler because there actually isn't a request to provide from the multi-region client.

**Comment by stobrien89** [comment]
Hi @pet0ruk, 

This has been addressed in #2426, which will be available in tomorrow's release.  We added a new method `useCustomHandler()` on the multiRegionClient, which allows you to pass custom handlers such as a MockHandler to the client.
